renku/core/commands/cluster.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017-2021 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku cluster command."""
+
+import click
+import os
+from pathlib import Path
+import requests
+from subprocess import PIPE, STDOUT, run
+from urllib.parse import quote
+
+from jinja2 import Template
+from pkg_resources import resource_filename
+
+from renku.core import errors
+from renku.core.management import LocalClient
+from renku.core.management.command_builder import inject
+from renku.core.management.command_builder.command import Command
+from renku.core.models.git import GitURL
+from renku.core.utils.urls import get_remote
+
+CONFIG_SECTION = "cluster"
+IMAGE_SHA = 7
+IMAGE_SUFFIX = "batch"
+DEFAULT_TEMPLATE = resource_filename("renku", "data/default_sbatch_template.sh")
+
+
+@inject.autoparams()
+def _prepare_git(client: LocalClient):
+    click.secho("Read Git configuration...", fg="yellow")
```

Comment:
the `renku/core/` namespace is used by both service and CLI and shouldn't know about click (as it's a CLI specific library).

We use the "magic" communication functionality to setup communications channels for core, e.g. https://github.com/SwissDataScienceCenter/renku-python/blob/858fe84ce2925a49d9b62638dc601f581e24353e/renku/core/commands/dataset.py#L458 in core and https://github.com/SwissDataScienceCenter/renku-python/blob/858fe84ce2925a49d9b62638dc601f581e24353e/renku/cli/dataset.py#L740-L741 setup in renku.cli